### ISSUE TITLE
MQ rules can be rendered "out-of-order" in v0.16

### ISSUE THREAD

**Issue by bryanrsmith** [FormidableLabs/radium#550]
Hello,

I've got a funny case with media queries that was broken in the 0.16 change to rendering media queries in a stylesheet. It happens when you have two identical media queries applied to the same element via separate style objects. One style object has media query rules at 300px and 400px, and a second style object has a media query rule at 300px. It looks like the rules in the two 300px media queries are being merged together and end up being rendered after the 400px MQ, so the 400px rule is never applied.

![image](https://cloud.githubusercontent.com/assets/186704/12523627/e33d921a-c10c-11e5-9e33-2014e5eccd1a.png)

It's a bit hard to explain, so here's a jsbin that reproduces the issue (used in the screenshot)
http://jsbin.com/jeseworifu/edit?js,output

This code was working as expected with 0.15.

Thanks a lot for your work on this library. It's been great to work with.

**Comment by ianobermiller** [comment]
Yeah, I figured this would happen, this is what happens when you use CSS :( We should be able to get around it by adding the CSS even when the media query is unused, then they'd always be added in the correct order.

**Comment by ianobermiller** [comment]
I think we'll be able to fix this in Radium itself, but for now you can work around it by giving the media query with `border: '5px solid hotpink'` a slightly different query (adding a space). It is, of course, hacky, but it prevents the two media queries (the one with the green bg and the one with the pink border) from being merged and applied later. See this riff on your bin: https://jsbin.com/pivipi/edit?js,output

@alexlande this is what happens when we start using `addCSS` more liberally :(

**Comment by bryanrsmith** [comment]
That's an easy workaround for now. Thanks!

**Comment by ianobermiller** [comment]
You're welcome. That actually gives me an idea for how to fix this; we would stop merging media queries, and instead just append a space to the key, so you'd end up with:

```js
{
  '@media (min-width: 300px)': { ... },
  '@media (min-width: 300px) ': { ... },
}
```

It isn't pretty, but it should work.

**Comment by tptee** [comment]
Looks like this didn't get built properly in the latest release. b29fdc3 isn't reflected in lib/merge-styles.js after an npm install...

**Comment by tptee** [comment]
This fix is now in `0.16.6`.
